third_party/move/move-command-line-common/src/testing.rs

```diff
@@ -63,3 +63,29 @@ pub fn format_diff(expected: impl AsRef<str>, actual: impl AsRef<str>) -> String
     }
     ret
 }
+
+pub fn format_diff_no_color(expected: impl AsRef<str>, actual: impl AsRef<str>) -> String {
```

Comment:
This is kind of redundant with the code above, but I can't simplify it much in Rust, so it's fine as is.  Higher-order functions in Rust are a bit verbose.

Here's a try:
```
pub fn format_diff_no_color(expected: impl AsRef<str>, actual: impl AsRef<str>) -> String {
    format_diff_shared(
        expected,
        actual,
        |x: &String, ret: &mut String| {
            ret.push_str(&format!("= {}", x.replace('\n', "\n= ")));
            ret.push('\n');
        },
        |x: &String, ret: &mut String| {
            ret.push_str(&format!("+ {}", x.replace('\n', "\n+ ")));
            ret.push('\n');
        },
        |x: &String, ret: &mut String| {
            ret.push_str(&format!("- {}", x.replace('\n', "\n- ")));
            ret.push('\n');
        },
    )
}

```

Where the code above is changed to:
```

fn format_diff_shared(
    expected: impl AsRef<str>,
    actual: impl AsRef<str>,
    same_proc: impl Fn(&String, &mut String),
    add_proc: impl Fn(&String, &mut String),
    rem_proc: impl Fn(&String, &mut String),
) -> String {
    use difference::{Changeset, Difference};

    let changeset = Changeset::new(expected.as_ref(), actual.as_ref(), "\n");

    let mut ret = String::new();

    for seq in changeset.diffs {
        match &seq {
            Difference::Same(x) => {
                same_proc(x, &mut ret);
            },
            Difference::Add(x) => {
                add_proc(x, &mut ret);
            },
            Difference::Rem(x) => {
                rem_proc(x, &mut ret);
            },
        }
    }
    ret
}

pub fn format_diff(expected: impl AsRef<str>, actual: impl AsRef<str>) -> String {
    format_diff_shared(
        expected,
        actual,
        |x: &String, ret: &mut String| {
            ret.push_str(x);
            ret.push('\n');
        },
        |x: &String, ret: &mut String| {
            ret.push_str("\x1B[92m");
            ret.push_str(x);
            ret.push_str("\x1B[0m");
            ret.push('\n');
        },
        |x: &String, ret: &mut String| {
            ret.push_str("\x1B[91m");
            ret.push_str(x);
            ret.push_str("\x1B[0m");
            ret.push('\n');
        },
    )
}
```